### PR TITLE
add template sync action to repo

### PR DESCRIPTION
**Pull request by JessicaS11**
One of the challenges for using this template across multiple events is trying to get per-event updates that should apply to the template back into the template. So far, this heroic work has been done by @jomey and @scottyhq on a largely manual basis. This PR adds the [template-sync](https://github.com/marketplace/actions/actions-template-sync) github action that will automatically (via cron on the 1st of the month) or manually open a PR in the templated repo that incorporates the changes made to the template. This will enable template-level updates to be made and merged directly into the template and then easily incorporated into any repos that were created using the template. This will be especially useful for upcoming events that will be sharing some tutorial content.

### CODE REVIEW

File: .github/actions/README.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ Quality assessment and quality control. Standardizes formatting including spell
 #### [repo2docker.yaml](../workflows/repo2docker.yaml)
 [Build a Docker image](https://github.com/jupyterhub/repo2docker-action) for JupyterHub/BinderHub
 
+#### [template-sync.yaml](../workflows/template-sync.yaml)
+Open a PR to update the templated repo to incorporate changes made to the
+[template repo](https://github.com/uwhackweek/jupyterbook-template).
+
 #### [test.yaml](../workflows/test.yaml)
 Build the websites (JupyterBook and front page). Run on Pull Requests against every commit and via a 'cron' schedule to maintain caching [since otherwise the cache expires if untouched in 7 days](https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy)
```
Comment (JessicaS11) [inline .github/actions/README.md:43]: Anyone know where this action went? repo2docker-PR.yaml is also not documented here.

Comment (jomey) [reply .github/actions/README.md:43]: I renamed "test" to build_website.yaml and (naughtily) did not update the docs
8e96762a61099b2a113ba72e681c4cc6ce295777

**Comment by jomey** [comment]
Nice find, @JessicaS11!

Think this will take off a lot of manual labor. One suggestion I have is that we create these PRs manually to begin with (via label) and see how it performs. When we do this manually, not all changes are transferred to the pages. For instance, I once swapped the sequence between schedule and team on the index.html for faster navigation during the event, but am not sure this should also be part of this template.

Maybe we can also do this the other way and specify which ones we want instead of saying which not?

**Comment by JessicaS11** [comment]
> One suggestion I have is that we create these PRs manually to begin with (via label) and see how it performs.

Sounds good. I commented out the cron for now, so the action will just have a manual trigger. 

> For instance, I once swapped the sequence between schedule and team on the index.html for faster navigation during the event, but am not sure this should also be part of this template.

In that case, wouldn't we just make the change in the templated repo instead of upstream in the template? The question then becomes whether or not to add index.html to the `.templatesyncignore` file for that event, but there's not an easy answer/solution when we get down to within-a-file changes. Either we'll have to add the file to the ignore list for that usage, or manually update the sync PR to not sync the entire file.

> Maybe we can also do this the other way and specify which ones we want instead of saying which not?

The action came with `.templatesyncignore` for config, but doesn't seem to have settings for specifying what we DO want synced. So using ignore versus want wasn't an intentional decision...

**Comment by jomey** [comment]
> In that case, wouldn't we just make the change in the templated repo instead of upstream in the template?

I guess I still have my head twisted on where the PR would come from. Pulling from the template to the event site? Then my described use case was not applicable. As you said, we can figure that out once we try this.

**Review by jomey — APPROVED** [review]

**Comment by JessicaS11** [comment]
> I guess I still have my head twisted on where the PR would come from.

My current understanding: you run the action in the templated (e.g. IS2, SnowEx) repo. The action checks THIS repo, and then opens a PR in whatever repo we've run the action in to port over any changes that were made here.

**Comment by jomey** [comment]
> The action checks THIS repo, and then opens a PR in whatever repo we've run the action in to port over any changes that were made here.

Then we should also exclude all the config .yaml files (`cookiecutter` and `book/_config`) since those might change with added/expanded website options.

**Comment by JessicaS11** [comment]
@scottyhq @jomey No PRs were getting the green button because main had a branch protection requiring "Build and test on OS - Ubuntu latest" to pass, but (I presume) because we weren't adding the preview label, the build wasn't being triggered, so its status could never be reported. I changed the rule to require "Build and test on OS - ${{matrix.os}}" to pass, but that just makes it "good to go" (since we didn't trigger that action?)... Do we just automatically need to add the preview label to all PRs and revert my branch rule change to avoid merging breaking/unbuildable changes?

**Comment by jomey** [comment]
> Do we just automatically need to add the preview label to all PRs and revert my branch rule change to avoid merging breaking/unbuildable changes?

The action only gets triggered once certain paths are changed. I am fine with keeping your rule updates since we are not needing a build for this kind of change, but will keep it for future book related paths updates.

**Comment by JessicaS11** [comment]
> The action only gets triggered once certain paths are changed.

Awesome! I didn't catch that and was worried it would let us merge even if a build didn't pass. Thanks!

**Pull request merged**